cli/test/test_run_parser.py

```diff
@@ -286,3 +287,152 @@ def test_identify_target_languages() -> None:
         translated=False, source_url="https://www.google.org", languages=["en"]
     )
     assert identify_translation_languages(doc_2, _target_languages) == set()
+
+
+@patch("cli.parse_pdfs.AzureApiWrapper.analyze_document_from_bytes")
+def test_fail_safely_on_azure_uncaught_exception(
+    mock_get, test_input_dir, caplog
+) -> None:
+    """
+    Test the functionality of the pdf parser.
+
+    Assert that we safely fail pdf parsing using the azure pdf parser should the default
+    endpoint fail with an uncaught Exception.
+    """
+    mock_get.side_effect = Exception(
+        mock.Mock(status=500), "Mock Internal Server Error"
+    )
+
+    with tempfile.TemporaryDirectory() as output_dir:
+        runner = CliRunner()
+
+        result = runner.invoke(
+            cli_main, [str(test_input_dir), output_dir, "--parallel"]
+        )
+
+        assert result.exit_code == 0
+
+        # Default config is to translate to English, and the HTML doc is already in
+        # English - so we just expect a translation of the PDF
+        assert set(Path(output_dir).glob("*.json")) == {
+            Path(output_dir) / "test_html.json",
+            Path(output_dir) / "test_pdf.json",
+            Path(output_dir) / "test_no_content_type.json",
+            Path(output_dir) / "test_pdf_translated_en.json",
+        }
+
+        for output_file in Path(output_dir).glob("*.json"):
+            parser_output = ParserOutput.parse_file(output_file)
+            assert isinstance(parser_output, ParserOutput)
+
+            if parser_output.document_content_type == CONTENT_TYPE_HTML:
+                assert parser_output.html_data.text_blocks not in [[], None]
+
+            if parser_output.document_content_type == CONTENT_TYPE_PDF:
+                assert parser_output.pdf_data.text_blocks in [[], None]
```

Comment:
I don't understand why these two checks are different ("not in" vs "in")

Reply:
We mock the response from the `analyze_document_from_bytes` function, this is the underlying function that the `analyze_large_document_from_bytes` also uses. 

Thus, we basically don't get a response for pdfs and thus we get no text blocks etc.